subprojects/docs/src/docs/userguide/organizingGradleProjects.adoc

```diff
@@ -117,13 +156,54 @@ A typical project including `buildSrc` has the following layout.
+For example you may want pass in a specific property value to control runtime behavior just for this one invocation of the build.
+Properties in a build script can easily become a maintenance headache and convolute the build script logic.
+The `gradle.properties` helps with keeping properties separate from the build script and should be explored as viable option.
+It's a good location for placing <<sec:gradle_configuration_properties,properties that control the build environment>>.
 
-A typical project setup places the `gradle.properties` file in the root directory of the build. Alternatively, the file can also live in the `GRADLE_USER_HOME` directory if you want to it apply to all builds on your machine.
+A typical project setup places the `gradle.properties` file in the root directory of the build.
+Alternatively, the file can also live in the `GRADLE_USER_HOME` directory if you want to it apply to all builds on your machine.
 
 ----
 .
 ├── build.gradle
 ├── gradle.properties
 └── settings.gradle
 ----
+
+=== Avoid writing to the same output directory from different tasks
+
+Tasks should define inputs and outputs to enable proper functioning of <<sec:up_to_date_checks,incremental build functionality>>.
+When declaring the outputs of a task, make sure that the directory for writing outputs is unique among all the tasks in your project.
+
+Intermingling or overwriting output files produced by different tasks compromises up-to-date checking causing slower builds.
+In turn, these filesystem changes may prevent Gradle's <<build_cache,build cache>> from properly identifying and caching what would otherwise be cacheable tasks.
+
+[[sec:custom_gradle_distribution]]
+=== Standardizing builds with a custom Gradle distribution
+
+<<init_scripts,Initialization scripts>> make it extremely easy to apply build logic across all projects on a single machine.
+Specifying an in-house binary repository and its credentials is a typical use case for setting up an initialization script.
+
+Often times enterprises want to standardize the build platform for all projects in the organization by defining common conventions or rules.
+You can achieve that with the help of initialization scripts.
+There are some drawbacks to the approach.
+First of all, you will have to communicate the setup process across all developers in the company.
+Furthermore, updating the initialization script logic uniformly can prove challenging.
+
+Custom Gradle distributions are a practical solution to this very problem.
+A custom Gradle distribution is comprised of the standard Gradle distribution plus one or many custom initialization scripts.
+The initialization scripts come bundled with the distribution and are applied every time the build is run.
+Developers only need to point their checked-in <<gradle_wrapper,Wrapper>> files to the URL of the custom Gradle distribution.
+
+Let's talk about the process for creating such a custom Gradle distribution. The following steps are required:
```

Comment:
Maybe just "The following steps are typical for creating a custom Gradle distribution:"?

Reply:
done